decidim-core/lib/decidim/map/provider/dynamic_map/here.rb

```diff
@@ -44,6 +44,58 @@ def javascript_snippets
               template.javascript_pack_tag("decidim_map_provider_here", defer: false)
             end
           end
+
+          private
+
+          # rubocop: disable Metrics/CyclomaticComplexity
```

Comment:
I think we should fix these Rubocop issues if you wrapped the language code mapping into a separate method as follows:
```ruby
def available_language_codes
  @available_language_codes ||= {
    "ar" => "ara",
    "eu" => "baq",
    "zn-ch" => "chi",
    # ...
  }
end
```

And then modified this method to something like:
```ruby
def language_code
  primary = I18n.locale.to_s
  secondary = primary.split("-")[0] # This will also try with only the 2-letter code e.g. for `pt-br` this would return `pt`
  available_language_codes[primary] || available_language_codes[secondary] || ""
end
```

This way you also wouldn't need to duplicate some of the languages in the original list if HERE doesn't support the language code.